===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(open_file:h)) at 465 (line 268, column 5) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/x264/input/avs.c
+++ b/table1/benchmarks/x264/input/avs.c
@@ -265,7 +265,10 @@ static int open_file( char *psz_filename, hnd_t *p_handle, video_info_t *info, c
     avs_hnd_t *h = calloc( 1, sizeof(avs_hnd_t) );
     if( !h )
         return -1;
-    FAIL_IF_ERROR( custom_avs_load_library( h ), "failed to load avisynth\n" );
+    if (custom_avs_load_library( h ))
+        x264_cli_log(NAME, X264_LOG_ERROR, "failed to load avisynth\n");
+        free(h);
+        return -1;
     h->env = h->func.avs_create_script_environment( AVS_INTERFACE_25 );
     if( h->func.avs_get_error )
     {
